pkg/authentication/externalidp/aws_client.go

```diff
@@ -0,0 +1,203 @@
+package authentication
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+	"github.com/treeverse/lakefs/pkg/api/apigen"
+)
+
+const (
+	AWSAuthVersion      = "2011-06-15"
+	AWSAuthMethod       = http.MethodPost
+	AWSAuthAction       = "GetCallerIdentity"
+	AWSAuthAlgorithm    = "AWS4-HMAC-SHA256"
+	StsGlobalEndpoint   = "sts.amazonaws.com"
+	AWSAuthActionKey    = "Action"
+	AWSAuthVersionKey   = "Version"
+	AWSAuthAlgorithmKey = "X-Amz-Algorithm"
+	//nolint:gosec
+	AWSAuthCredentialKey = "X-Amz-Credential"
+	AWSAuthDateKey       = "X-Amz-Date"
+	AWSAuthExpiresKey    = "X-Amz-Expires"
+	//nolint:gosec
+	AWSAuthSecurityTokenKey  = "X-Amz-Security-Token"
+	AWSAuthSignedHeadersKey  = "X-Amz-SignedHeaders"
+	AWSAuthSignatureKey      = "X-Amz-Signature"
+	AWSDatetimeFormat        = "20060102T150405Z"
+	AWSCredentialTimeFormat  = "20060102"
+	AWSDefaultSTSLoginExpire = 15 * time.Minute
+)
+
+var ErrAWSCredentialsExpired = errors.New("AWS credentials expired")
+var ErrRetrievingToken = errors.New("failed to retrieve token")
+
+type LoginResponse struct {
+	Token string
+}
+
+type IDPProvider interface {
```

Comment:
IDP == Identity Provider -> IDPProvider == IdentityProviderProvider 😋 

```suggestion
type Provider interface {
```

Since the package name contains the name idp and external, in go you don't need to duplicate the name, i.e this is enough: 

```go
import (
    ".../externalidp"
)

// usage is clear
externalidp.Provider
```